ddmrp_history/models/stock_warehouse_orderpoint.py

```diff
@@ -112,8 +122,113 @@ def stacked(df, categories):
                    line_dash='dotted')
 
             script, div = components(p)
-            rec.history_chart = '%s%s' % (div, script)
+            rec.planning_history_chart = '%s%s' % (div, script)
 
-    # TODO: computed through a button? able to select period.
-    history_chart = fields.Text(
-        string='Historical Chart', compute='_compute_history_chart')
+    def _compute_execution_history_chart(self):
+        start_stack = 0
+
+        def stacked(df, categories):
+            areas = dict()
+            last = np.zeros(len(df[categories[0]]))
+            last += start_stack
+            for cat in categories:
+                next = last + df[cat]
+                areas[cat] = np.hstack((last[::-1], next))
+                last = next
+            return areas
+
+        for rec in self:
```

Comment:
save some space by declaring at beginning of the loop:
```python
history_model = self.env['ddmrp.history']
domain = [('orderpoint_id', '=', rec.id)]
```

Reply:
Attended here : https://github.com/Eficent/ddmrp/pull/52/commits/cb36fca16a87a1623e522c03917af78a587eb4c1